src/shared/environment-data-sources/foxy.ts

```diff
@@ -0,0 +1,99 @@
+                                    item_category = product.item_category,
+                                    price = product.price,
+                                    quantity = product.quantity;
+
+                                return {
+                                    orderId: transaction_id.toString(),
+                                    sku: item_id.toString(),
+                                    name: item_name?.toString() || "N/A",
+                                    category: item_category?.toString() || "N/A",
+                                    unitPrice: parseFloat(price || 0),
+                                    quantity: parseInt(quantity || 1),
+                                    currency: "USD",
+                                } as TransactionCartItem;
+                            }),
+                        }
+                    });
+
+                    break;
+                }
+                if (dataCart.event === "purchase") {
+                    const transaction = dataCart.ecommerce;
+                    const products = transaction.items;
+                    const { transaction_id, value, tax, shipping } = transaction;
+
+                    observable.notify({
+                        transactionEvent: {
+                            total: parseFloat(value || 0) + parseFloat(tax || 0) + parseFloat(shipping || 0),
+                            id: transaction_id.toString(),
+                            tax: parseFloat(tax) || 0,
+                            shipping: parseFloat(shipping) || 0,
+                            city: "N/A",
+                            state: "N/A",
+                            country: "N/A",
+                            currency: "USD",
+                            items: products.map((product) => {
+                                var item_id = product.item_id,
+                                    item_name = product.item_name,
+                                    item_category = product.item_category,
+                                    price = product.price,
+                                    quantity = product.quantity;
+
+                                return {
+                                    orderId: transaction_id.toString(),
+                                    sku: item_id.toString(),
+                                    name: item_name?.toString() || "N/A",
+                                    category: item_category?.toString() || "N/A",
+                                    unitPrice: parseFloat(price || 0),
+                                    quantity: parseInt(quantity || 1),
+                                    currency: "USD",
+                                } as TransactionCartItem;
+                            }),
+                        }
+                    });
+
+                    break;
+                }
+            }
+        }
+    }
+
+    datalayerSource(() => {
+        runFoxy();
+    });
+};
```

Comment:
Hello, why are we using datalayersource here if we're not getting data from the data layer?